packages/aragon-wrapper/src/index.js

```diff
@@ -263,11 +291,24 @@ export default class Aragon {
       )
       .flatMap(
         (apps) => Promise.all(
-          apps.map(async (app) => Object.assign(
-            app,
-            await this.apm.getLatestVersionForContract(app.appId, app.codeAddress)
-              .catch(() => getAragonOsInternalAppInfo(app.appId)) // for internal apps we check local mapping
-          ))
+          apps.map(async (app) => {
+            const cacheKey = `${app.appId}.${app.codeAddress}`
+            const cachedAppInfo = dotprop.get(appInfoCache, cacheKey)
+
+            const appInfo =
+              cachedAppInfo ||
+              getAragonOsInternalAppInfo(app.appId) ||
+              (await this.apm
+                .getLatestVersionForContract(app.appId, app.codeAddress)
+                // Just silence any errors
+                .catch(() => {}))
+
+            if (!cachedAppInfo && appInfo) {
+              dotprop.set(appInfoCache, cacheKey, appInfo)
```

Comment:
Isn’t `appInfoCache` going to grow indefinitely?

Reply:
It will, based on how many apps (combinations of `appId` and `base contract address`) you install. I don't think we need to worry about this, as it'll likely stay very low (on current DAOs this will be 5-6), but we can start thinking about a cache invalidation strategy once it's a problem.

Reply:
Yes I don’t think it would be a problem right now, but I was more thinking about the app running for a very long time, browsing several DAOs (e.g. someone having a pinned tab in the browser only for Aragon). A strategy could be to use a fixed-size array instead of an object.

Reply:
Made https://github.com/aragon/aragon.js/issues/181 for this, let's worry about it later :).